File: LeetCode/Java Solutions/Check If Two String Arrays are Equivalent/check.java
public class Solution {
    public boolean arrayStringsAreEqual(String[] word1, String[] word2) {
        // Use StringBuilder for efficient string concatenation
        StringBuilder concatenatedWord1 = new StringBuilder();
        StringBuilder concatenatedWord2 = new StringBuilder();

        // Append all strings from word1 to concatenatedWord1
        for (String s : word1) {
            concatenatedWord1.append(s);
        }

        // Append all strings from word2 to concatenatedWord2
        for (String s : word2) {
            concatenatedWord2.append(s);
        }

        // Compare the two concatenated strings
        return concatenatedWord1.toString().equals(concatenatedWord2.toString());
    }
}

File: LeetCode/Python Solutions/Check If Two String Arrays are Equivalent/check.py
class Solution:
    def arrayStringsAreEqual(self, word1: List[str], word2: List[str]) -> bool:
        # Concatenate all strings in word1
        concatenated_word1 = ''.join(word1)
        # Concatenate all strings in word2
        concatenated_word2 = ''.join(word2)

        # Compare the two concatenated strings
        return concatenated_word1 == concatenated_word2
